check-version.sh

```diff
@@ -10,29 +10,39 @@ trap 'echo "$0: Check failed. Stopping." >&2' ERR
 readonly version_path='VERSION'
 readonly changes_path='doc/source/changes.rst'
 
+function check_status() {
+    git status -s "$@"
+    test -z "$(git status -s "$@")"
+}
+
+function get_latest_tag() {
+    local config_opts
+    printf -v config_opts ' -c versionsort.suffix=-%s' alpha beta pre rc RC
```

Comment:
`printf -v` is a `bash`-specific feature, made possible by `printf`'s status as a builtin in `bash`. Since it does its work in the shell process, it can write to state in the shell process, including writing shell variables. In most other Bourne-style shells, and when only POSIX features can be assumed, `printf -v` is not available, and `printf` need not even be provided as a `builtin` (as with `echo`, POSIX requires that an external `printf` command be available).